### PR TITLE
Add extended JSON encoder for datetime objects in message encoding

### PR DESCRIPTION
**Pull request by MHHukiewitz**
Supersedes https://github.com/aleph-im/aleph-client/pull/131

### CODE REVIEW

**Review by 1yam — APPROVED** [review]
LGTM

File: src/aleph/sdk/client/authenticated_http.py [hunk 1]
```diff
@@ -50,6 +51,16 @@
     magic = None  # type:ignore
 
 
+def extended_json_encoder(obj: Any) -> str:
+    if (
+        isinstance(obj, datetime.datetime)
+        or isinstance(obj, datetime.date)
+        or isinstance(obj, datetime.time)
+    ):
+        return obj.isoformat()  # or any other format you prefer
```
Comment (hoh) [inline src/aleph/sdk/client/authenticated_http.py:60]: "or any other format you prefer" ?

Comment (hoh) [reply src/aleph/sdk/client/authenticated_http.py:60]: Is the timezone included ?

File: tests/unit/test_asynchronous.py [hunk 2]
```diff
@@ -121,3 +124,63 @@ async def test_forget(mock_session_with_post_success):
...
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "message_type, content",
+    [
+        (
+            MessageType.aggregate,
+            {
+                "content": {"Hello": datetime.datetime.now()},
```
Comment (hoh) [inline tests/unit/test_asynchronous.py:136]: What about timezones ?

Comment (MHHukiewitz) [reply tests/unit/test_asynchronous.py:136]: Yeah, should check that too

**Pull request merged**